client/components/telecom/telephony/bulkAction/modal/telephony-bulk-action-modal.controller.js

```diff
@@ -174,24 +176,37 @@ angular.module("managerApp").controller("telephonyBulkActionModalCtrl", function
                 serviceType: self.bindings.serviceType
             }).value();
 
-            // apply custom filter if provided
-            if (self.bindings.filterServices && _.isFunction(self.bindings.filterServices())) {
-                allServices = self.bindings.filterServices()(allServices);
+            // custom filter asynchronous
+            if (self.bindings.filterServicesAsync && _.isFunction(self.bindings.filterServicesAsync())) {
```

Comment:
Okay for a async filter. But shouldn't we use `filterServices` by testing if it returns a `promise` instead of creating a new function callback for the component ?